### PR TITLE
Curve metapool fork

### LINKED ISSUE #36: provide XUSD -> individual coin capability

### PR DESCRIPTION
**Pull request by 0xKirkjufell**
Dapp change checklist:
  - [x] Code reviewed by 1 reviewers.
  - [x] CI passes

closes #36

### CODE REVIEW

File: src/hooks/useCurrencySwapper.js [hunk 1]
```diff
@@ -235,6 +248,81 @@ const useCurrencySwapper = ({
...
+
+    return {
+      result: await _swapCurve(swapAmount, minSwapAmountReceived, false),
+      swapAmount,
+      minSwapAmount,
+    }
+  }
+
+  const _maybeToAvToken = (address) => {
+    // TODO: do this not in a stupid way, perhaps using coins and underlying coins call?
```
Comment (0xKirkjufell) [inline src/hooks/useCurrencySwapper.js:299]: Can you think of a better way to do this here? I do not like this.

1) put addresses in contractAddresses.js and pull in by name
2) zip curveXUSDMetaPoolUnderlyingCoins to the contracts using the known indexes
3) ?

File: src/hooks/useSwapEstimator.js [hunk 2]
```diff
@@ -269,6 +277,71 @@ const useSwapEstimator = ({
...
+        error: 'unsupported',
+      }
+    }
+
+    try {
+      const priceQuoteBn = await quoteCurve(swapAmount)
+      const amountReceived = ethers.utils.formatUnits(
+        priceQuoteBn,
+        // 18 because xusd has 18 decimals
+        isRedeem ? coinToReceiveDecimals : 18
```
Comment (0xKirkjufell) [inline src/hooks/useSwapEstimator.js:296]: pull from coinInfoList

File: src/stores/ContractStore.js [hunk 3]
```diff
@@ -59,6 +59,7 @@ const ContractStore = new Store({
   isGasPriceUserOverriden: false,
   readOnlyProvider: false,
   showAllContracts: false,
+  curveMetapoolUnderlyingCoins: false,
```
Comment (0xKirkjufell) [inline src/stores/ContractStore.js:62]: idiom: `false` or `[]`?

File: src/utils/contracts.js [hunk 4]
```diff
@@ -15,6 +15,120 @@ import usdtAbi from 'constants/mainnetAbi/usdt.json'
...
+    outputs: [
+      {
+        name: '',
+        type: 'address[8]',
+      },
+    ],
+  },
+]
+
+const curveMetapoolMiniAbi = [
```
Comment (0xKirkjufell) [inline src/utils/contracts.js:38]: unused

File: src/hooks/useCurrencySwapper.js (hunk 1 above)
Comment (0xAnnapurna) [reply src/hooks/useCurrencySwapper.js:299]: I think we should create a single coin info struct and put this translation in there as e.g. `curveZapperAvTokenAddress`

**Review by 0xAnnapurna — APPROVED** [review]

File: src/hooks/useCurrencySwapper.js (hunk 1 above)
Comment (0xKirkjufell) [reply src/hooks/useCurrencySwapper.js:299]: created issue

File: src/stores/ContractStore.js (hunk 3 above)
Comment (0xAnnapurna) [reply src/stores/ContractStore.js:62]: Should be `[]`

**Pull request merged**